pkgs/os-specific/darwin/apple-source-releases/libffi/package.nix

```diff
@@ -64,6 +66,9 @@ mkAppleDerivation (finalAttrs: {
     # The Darwin SDK puts the headers in `include/ffi`. Add a symlink for compatibility.
     ''
       ln -s "$dev/include" "$dev/include/ffi"
+      # Make sure Apple’s header with availability annotations is installed in place of the generated one.
+      # Use `macCatalyst` instead of `iosmac` to avoid errors due to invalid availability annotations.
+      substitute darwin/include/ffi.h "$dev/include/ffi.h" --replace-fail iosmac macCatalyst
```

Comment:
By the way, is this an internal SDK difference or a bootstrap tools thing or what?

Reply:
The header that Apple ships uses `macCatalyst`, but the one in the source release uses `iosmac`. I assume they do some sort of post-processing when they build the SDK.